### PR TITLE
M2-3348: Add a checkbox that enables streaming of response data

### PR DESCRIPTION
**Pull request by nmialik**

### CODE REVIEW

File: src/modules/Builder/features/BuilderAppletSettings/BuilderAppletSettings.utils.tsx [hunk 1]
```diff
@@ -25,33 +26,38 @@ export const getSettings = ({
...
-                param: SettingParam.DataRetention,
-                disabled: isNewApplet,
-                tooltip,
-                'data-testid': `${dataTestid}-data-retention`,
-              },
-            ],
-          },
-        ]
-      : []),
+    {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/BuilderAppletSettings/BuilderAppletSettings.utils.tsx:29]: don't we need isManagerOrOwner check anymore?

File: src/shared/components/NavigationMenu/NavigationMenu.types.ts [hunk 2]
```diff
@@ -8,12 +8,14 @@ export type NavigationItem = {
   param: string;
   tooltip?: string;
   disabled?: boolean;
+  visibility?: boolean;
```
Comment (IhorNesterenko99) [inline src/shared/components/NavigationMenu/NavigationMenu.types.ts:11]:
```suggestion
  isVisible?: boolean;
```

File: src/shared/components/NavigationMenu/NavigationMenu.types.ts [hunk 3]
```diff
@@ -8,12 +8,14 @@ export type NavigationItem = {
...
   'data-testid'?: string;
 };
 
 export type Item = {
   label: string;
   items: NavigationItem[];
+  visibility?: boolean;
```
Comment (IhorNesterenko99) [inline src/shared/components/NavigationMenu/NavigationMenu.types.ts:18]:
```suggestion
 isVisible?: boolean;
```

File: src/shared/components/NavigationMenu/LeftBar/LeftBar.tsx [hunk 4]
```diff
@@ -27,41 +27,49 @@ export const LeftBar = ({ title, items, hasActiveItem, onItemClick }: LeftBarPro
...
+                      icon,
+                      label,
+                      component,
+                      param,
+                      disabled,
+                      tooltip,
+                      visibility: itemVisibility = true,
+                      'data-testid': dataTestid,
+                    }) =>
+                      itemVisibility && (
```
Comment (IhorNesterenko99) [inline src/shared/components/NavigationMenu/LeftBar/LeftBar.tsx:47]:
```suggestion
                      isItemVisible && (
```

Comment (IhorNesterenko99) [inline src/shared/components/NavigationMenu/LeftBar/LeftBar.tsx:47]: maybe we need ternary here?

File: src/shared/features/AppletSettings/LiveResponseStreamingSetting/LiveResponseStreamingSetting.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { useFormContext } from 'react-hook-form';
+import { Trans, useTranslation } from 'react-i18next';
+
+import { CheckboxController } from 'shared/components/FormComponents';
+import { StyledAppletSettingsDescription } from 'shared/features/AppletSettings/AppletSettings.styles';
```
Comment (IhorNesterenko99) [inline src/shared/features/AppletSettings/LiveResponseStreamingSetting/LiveResponseStreamingSetting.tsx:5]: could you pls export AppletSettings.styles from index?

File: src/shared/components/NavigationMenu/LeftBar/LeftBar.tsx (hunk 4 above)
Comment (nmialik) [inline src/shared/components/NavigationMenu/LeftBar/LeftBar.tsx:47]: why?

File: src/shared/components/NavigationMenu/NavigationMenu.types.ts (hunk 2 above)
Comment (kdanilenka32) [reply src/shared/components/NavigationMenu/NavigationMenu.types.ts:11]: visibility is is ok)

**Review by kdanilenka32 — APPROVED** [review]

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by muhurov — APPROVED** [review]

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**